### DISCUSSION TITLE
Folder converting to submodule in Repo

### DISCUSSION THREAD

**Discussion by Deepaktenzi** [community/community#29405] [New to GitHub] [answered]
Hello everyone.! .
Can someone please tell me how to add client and server folders in the repo...Only one folder is uploaded after that one folder is not opening it's showing an arrow ->.....

Can anyone tell me how to do it..

![Screenshot 2022-08-14 at 2 40 31 PM](https://user-images.githubusercontent.com/39908306/184530229-96f649e5-9540-4afa-b63b-1921bb2e0470.png)
.

**Comment by airtower-luna** [comment, marked as answer]
That arrow symbol indicates a submodule. It looks like you accidentally turned the `client` folder into another Git repository instead of just adding the files inside.

**Reply by Deepaktenzi** [reply]
How to resolve this issue @airtower-luna  ..this is my first Git repository

**Reply by airtower-luna** [reply]
If you didn't mean to turn `client` into a Git repository, just delete the hidden `.git` directory *inside it*. Do **not** delete the one of the enclosing directory that you want to be a Git repository. :warning:

Afterwards your Git repository should see `client` like any other directory. Add, commit, and push the files inside. :slightly_smiling_face:

**Reply by Deepaktenzi** [reply]
Thanks, it's work...

**Reply by Peterking1972** [reply]
cando   te  encenbtres  desesperado  por    algna   cuestion  en  particular. Solo camina  lo  mas  h<sta   que   saques toda   la  energia  negativa  con  la   que  centas  dentro  de tu  cuerpo  , mente  y  alma, .   Ya   que  la  energia   que capturamos  en  la   vida   cotidiana. Es  la  que  nos  hace proyectarnoa  sanmente.

**Reply by romdhanikhalil16** [reply]
@airtower-luna  please How to fix it when deleting the local project ? I only have the github version

**Reply by airtower-luna** [reply]
If you mean you want to use the GitHub web interface, you can't do it. You need a local copy (clone) of the repository. An online development environment (like Codespaces) might work, too, if it allows you to work with the filesystem and use Git commands.

However, if you want to replace a broken submodule with a directory that contains files you still need the actual files.

**Reply by Prakhaxx** [reply]
@airtower-luna @Deepaktenzi I am facing the same issue, but I am not able to understand the solution given above please tell me in a simpler way because I am new to git and GitHub.

**Reply by Deepaktenzi** [reply]
@Prakhaxx You have to go into your client directory ( Or Any folder which is showing an arrow) and delete the .git file it's a hidden file.
run this command in the client directory  run this command `rm -rf .git`

**Reply by ghost** [reply]
rm -rf .git done, still same issue

**Reply by airtower-luna** [reply]
After deleting the inner repository (which the outer repository would have treated as a submodule) the outer repository should show the files inside as new, and the submodule as deleted. Double-check it's as you want, then commit and push.

**Reply by anurag-sachan** [reply]
> remove already initialised .git file from subdirectory
```
rm -rf .git
```

> remove cached subdirectory
```
git rm --cached <subdirectory>
```

> add, commit & push folder again.

In this case,
1. cd client
2. rm -rf .git
3. git rm --cached client
4. git add, commit, push.

**Reply by Deepaktenzi** [reply]
The folder which is Submodule (arrow Folder) 
```
1: cd to that Directory
2: Then run  " rm -rf .git " ( which will delete the .git file from the folder (It's a hidden file)
```

Suppose you have two folders in project CLIENT and SERVER then don't initialize Git in each folder just Initialise a Git repository to your main project folder. It will resolve the issues. 

```
add 👍🏻 
commit 👍🏻 
push 🥇 
```

**Reply by sid-cmd19** [reply]
Thanks @anurag-sachan and everyone. This was informative and helpful

**Reply by Asadrana123** [reply]
But sir I have deleted this project from my pc so what i can do now?? please tell !!

**Reply by BilelJedidi** [reply]
I did everything, deleted the .git hidden file then when I add commit and push I get that every thing is up-to-date and nothings changes in the git repo

**Reply by Omkarbhandia** [reply]
rm -rf .git command is giving me error "Remove-Item : A parameter cannot be found that matches parameter name 'rf'"
 Please help with this @anurag-sachan @Deepaktenzi

**Reply by amol01netke** [reply]
Thank you

**Comment by abhishekkhanna2710** [comment]
No changes give the correct solution please

**Reply by kamalraj21intel** [reply]
I got help from this thread https://stackoverflow.com/questions/62056294/github-folders-have-a-white-arrow-on-them

**Reply by anurag-sachan** [reply]
1. cd client
    2. rm -rf .git
    3. git rm --cached client
    4. git add, commit, push.

**Reply by Deepaktenzi** [reply]
The folder which is Submodule (arrow Folder) 
```
1: cd to that Directory
2: Then run  " rm -rf .git " ( which will delete the .git file from the folder (It's a hidden file)
```

Suppose you have two folders in project CLIENT and SERVER then don't initialize Git in each folder just Initialise a Git repository to your main project folder. It will resolve the issues. 

```
add 👍🏻 
commit 👍🏻 
push 🥇 
```

**Reply by joelgithire** [reply]
1. cd client
2. rm -rf .git
3. remane client to another name
4. git add, commit , push

**Reply by Audiopolis** [reply]
Deleted .git directory. It does not exist anymore. Still shows as a submodule.

**Comment by MoodJyothirmai** [comment]
<img width="465" alt="Folder Name" src="https://github.com/user-attachments/assets/8fc16ffc-b265-43e2-96b6-65fba73d2d08">

Why this directory names are appearing side by side when I created Day1_08082024 inside the directory of src  directory?

Can anyone help me out.

**Reply by amol01netke** [reply]
check whether src is uppercase or lowercase
